### ISSUE TITLE
torchvision ImageFolder filepath type issue

### ISSUE THREAD

**Issue by wztdream** [microsoft/pyright#2028]
ImageFolder suport Path type, but the type annotation of parameter root is 'str', so will trigger warning:
`Argument of type "Path" cannot be assigned to parameter "root' of type "str"...`

Basically this should be an issue with torchvsion as the type annotation should be `root:str|Path`, I post this issue here to see whether pyright can do anything with this issue.

```python
from pathlib import Path

from torchvision.datasets import ImageFolder

a = Path(".")

dataset = ImageFolder(root=a) # here will be warning

```

**Comment by erictraut** [comment]
Pyright is correct to flag an error here. If the `root` parameter accepts `Path`, then it's a bug in the torchvision library and needs to be fixed there.

**Comment by wztdream** [comment]
I see, but as treating path as a string is very common and pathlib is also common used, so is there any method to ignore this warning manually in pyright before torchvision or pathlib upstream fixed the bug?

**Comment by wztdream** [comment]
@erictraut  You can imagine that to silence this warning totally, we need to update **all codes** that related with file path, update `path:str` to 'path:str|Path`, this seems impossible. Maybe we need special handle `Path` and treat it as a string in pyright?

**Comment by erictraut** [comment]
If this is a bug in the library, it needs to be fixed there. Please file a bug report for the library, or better yet submit a PR with a fix. It doesn't make sense for a type checker to incorporate work-arounds for bugs in third-party libraries.
